sentry-types/src/project_id.rs

```diff
@@ -8,88 +7,45 @@ use thiserror::Error;
 /// Raised if a project ID cannot be parsed from a string.
 #[derive(Debug, Error, PartialEq, Eq, PartialOrd, Ord)]
 pub enum ParseProjectIdError {
-    /// Raised if the value is not an integer in the supported range.
-    #[error("invalid value for project id")]
-    InvalidValue,
     /// Raised if an empty value is parsed.
     #[error("empty or missing project id")]
     EmptyValue,
 }
 
 /// Represents a project ID.
-#[derive(Copy, Clone, Debug, PartialEq, Eq, Ord, PartialOrd, Hash, Deserialize, Serialize)]
-pub struct ProjectId(u64);
+#[derive(Clone, Debug, PartialEq, Eq, Ord, PartialOrd, Hash, Deserialize, Serialize)]
+pub struct ProjectId(String);
 
 impl ProjectId {
-    /// Creates a new project ID from its numeric value.
+    /// Creates a new project ID from its string representation.
+    /// This assumes that the string is already well-formed and URL
+    /// encoded/decoded.
     #[inline]
-    pub fn new(id: u64) -> Self {
-        Self(id)
+    pub fn new(id: &str) -> Self {
+        Self(id.to_string())
     }
 
-    /// Returns the numeric value of this project id.
+    /// Returns the string representation of the project ID.
     #[inline]
-    pub fn value(self) -> u64 {
-        self.0
+    pub fn value(&self) -> &str {
+        self.0.as_ref()
     }
 }
 
 impl fmt::Display for ProjectId {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        write!(f, "{}", self.value())
+        write!(f, "{}", self.0)
     }
 }
 
-macro_rules! impl_from {
-    ($ty:ty) => {
-        impl From<$ty> for ProjectId {
-            #[inline]
-            fn from(val: $ty) -> Self {
-                Self::new(val as u64)
-            }
-        }
-    };
-}
-
-impl_from!(u8);
```

Comment:
IMO, we could keep the Try/From impls, as they would just stringify the given int. But its also fine to just remove it.

Reply:
i'll follow up on this in a different place, but i'm curious to hear the argument for keeping the Try/From impls. i've got an inkling of an idea, but it could just be completely wrong. 

i'm open to adding these back in a separate PR though. right now it feels like keeping impls just for integer values looks a little weird: why are there `Try/From` impls for integers but `parse` for `&str`s? why aren't there `try_from` impls for other primitives, why specifically all of the ints?